rts/Game/UnsyncedGameCommands.cpp

```diff
@@ -3833,85 +3875,29 @@ class CommandHelpActionExecutor : public IUnsyncedActionExecutor {
+bool UnsyncedGameCommands::ActionPressed(const Action& action, bool isRepeat)
 {
-	switch (hashString(action.command.c_str())) {
-		case hashString("drawinmap"): {
-			inMapDrawer->SetDrawMode(false);
-		} break;
-
-		case hashString("moveforward"): {
-			camera->SetMovState(CCamera::MOVE_STATE_FWD, false);
-		} break;
-		case hashString("moveback"): {
-			camera->SetMovState(CCamera::MOVE_STATE_BCK, false);
-		} break;
-		case hashString("moveleft"): {
-			camera->SetMovState(CCamera::MOVE_STATE_LFT, false);
-		} break;
-		case hashString("moveright"): {
-			camera->SetMovState(CCamera::MOVE_STATE_RGT, false);
-		} break;
-		case hashString("moveup"): {
-			camera->SetMovState(CCamera::MOVE_STATE_UP, false);
-		} break;
-		case hashString("movedown"): {
-			camera->SetMovState(CCamera::MOVE_STATE_DWN, false);
-		} break;
-
-		case hashString("movefast"): {
-			camera->SetMovState(CCamera::MOVE_STATE_FST, false);
-		} break;
-		case hashString("moveslow"): {
-			camera->SetMovState(CCamera::MOVE_STATE_SLW, false);
-		} break;
-		case hashString("movetilt"): {
-			camera->SetMovState(CCamera::MOVE_STATE_TLT, false);
-		} break;
-		case hashString("movereset"): {
-			camera->SetMovState(CCamera::MOVE_STATE_RST, false);
-		} break;
-		case hashString("moverotate"): {
-			camera->SetMovState(CCamera::MOVE_STATE_RTT, false);
-		} break;
+	const IUnsyncedActionExecutor* executor = GetActionExecutor(action.command);
 
-		case hashString("mouse1"): {
-			mouse->MouseRelease(mouse->lastx, mouse->lasty, 1);
-		} break;
-		case hashString("mouse2"): {
-			mouse->MouseRelease(mouse->lastx, mouse->lasty, 2);
-		} break;
-		case hashString("mouse3"): {
-			mouse->MouseRelease(mouse->lastx, mouse->lasty, 3);
-		} break;
+	if (executor != nullptr) {
+		// an executor for that action was found
+		if (executor->ExecuteAction(UnsyncedAction(action, isRepeat)))
+			return true;
+	}
 
-		#if 0
-		// HACK   somehow weird things happen when MouseRelease is called for button 4 and 5.
-		// Note that SYS_WMEVENT on windows also only sends MousePress events for these buttons.
-		case hashString("mouse4"): {
-			mouse->MouseRelease(mouse->lastx, mouse->lasty, 4);
-		} break;
-		case hashString("mouse5"): {
-			mouse->MouseRelease(mouse->lastx, mouse->lasty, 5);
-		} break;
-		#endif
+	return false;
+}
 
-		case hashString("mousestate"): {
-			mouse->ToggleMiddleClickScroll();
-		} break;
-		case hashString("gameinfoclose"): {
-			CGameInfo::Disable();
-		} break;
+bool UnsyncedGameCommands::ActionReleased(const Action& action)
+{
+	const IUnsyncedActionExecutor* executor = GetActionExecutor(action.command);
 
-		default: {
-		} break;
+	if (executor != nullptr) {
+		// an executor for that action was found
+		executor->ExecuteActionRelease(UnsyncedAction(action, false));
 	}
 
-	return false;
+	return false; // prior implementation always returned false - I don't know why, but we'll maintain behaviour until we have reason to change it.
```

Comment:
Notice all previous release inline executed actions were about setting a modifier state update.

The question then becomes do we want to take this opportunity for refactor to implement the fallback behavior consistently (instead of just always returning false?).

If the answer is yes, and I think it makes sense, then on each action extracted from the big switch statement above we return false (all previous release actions didn't halt the action chain) but here we return true when any of the executors returns true, makes sense? Let me know your thoughts.